Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Use the official Golang image to build the Go binary
+FROM golang:1.22 AS builder
+
+# Set the Current Working Directory inside the container
+WORKDIR /app
+
+# Copy go.mod and go.sum files and download dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the source code into the container
+COPY . .
+
+# Build the Go app
+RUN GOOS=linux GOARCH=amd64 go build -o main cmd/productivite/main.go
+
+# Use a minimal base image to run the Go binary
+FROM alpine:latest
+
+# Install necessary packages
+RUN apk --no-cache add ca-certificates
+
+# Set the Current Working Directory inside the container
+WORKDIR /root/
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/aakashgusain1990/productivite/pull/12#issuecomment-2426201042>#bb8fd9</a><span> - 10/21/2024, 03:34 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The Dockerfile sets the working directory to /root/ and runs the application as root, which is a security risk.
</div>
<div id="fix">
<b>Fix:</b> Create a non-root user in the Dockerfile and switch to this user before running the application.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -18,7 +18,11 @@
  FROM alpine:latest
 
  # Install necessary packages
 -RUN apk --no-cache add ca-certificates
 +RUN apk --no-cache add ca-certificates shadow
 +
 +# Create a non-root user
 +RUN adduser -D appuser
 +USER appuser
 
  # Set the Current Working Directory inside the container
 -WORKDIR /root/
 +WORKDIR /home/appuser/
 @@ -27,10 +31,10 @@
  COPY --from=builder /app/main .
 
  # Copy the pre-built binary from the builder stage
 -COPY --from=builder /app/config.yaml config.yaml
 +COPY --from=builder --chown=appuser:appuser /app/config.yaml config.yaml
 
  # Ensure the binary is executable
 -RUN chmod +x /root/main
 +RUN chmod +x /home/appuser/main
 
  # Debug: Check if the binary exists and is executable
 -RUN ls -l /root/main
 +RUN ls -l /home/appuser/main
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged